android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt

```diff
@@ -13,23 +14,38 @@ import io.intercom.android.sdk.UnreadConversationCountListener
 import io.intercom.android.sdk.UserAttributes
 import io.intercom.android.sdk.identity.Registration
 import io.intercom.android.sdk.push.IntercomPushClient
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
 
-class IntercomFlutterPlugin(private val application: Application) : MethodCallHandler, EventChannel.StreamHandler {
-
-
+class IntercomFlutterPlugin : FlutterPlugin, MethodCallHandler, EventChannel.StreamHandler, ActivityAware {
   companion object {
+    @JvmStatic lateinit var application: Application
+
     @JvmStatic
     fun registerWith(registrar: Registrar) {
       val channel = MethodChannel(registrar.messenger(), "maido.io/intercom")
-      channel.setMethodCallHandler(IntercomFlutterPlugin(registrar.context() as Application))
+      application = registrar.context() as Application
+      channel.setMethodCallHandler(IntercomFlutterPlugin()
```

Comment:
```suggestion
      channel.setMethodCallHandler(IntercomFlutterPlugin())
```

Missing closing parenthesis